### DISCUSSION TITLE
Add to a way:layer the start and end of a way that meets tag criteria

### DISCUSSION THREAD

**Discussion by BenOnTrack** [systemed/tilemaker#513] [Q&A] [answered]
Hi, I'd like to know if tilemaker currently host function that would allow, while processing ways, to add to a way:Layer the start and end of the way ? I can't find any function that "work"on geometry, besides the "LayerAsCentroid" that can compute the centroid of a way (ideal for POI).

My current use-case is to compute start and end of specific ways, such as "hiking trail".
My current workaround is to work with osmium and post-process the pbf to generate a mbtiles that host all those start and end, but that'd be way more efficient if that could be performed directly while generating the tiles.

**Comment by systemed** [comment, marked as answer]
It doesn’t at present, but it shouldn’t be hard to add. Would you like to submit an issue for it?

**Reply by BenOnTrack** [reply]
thanks for the quick feedback -> https://github.com/systemed/tilemaker/issues/514 (hope that matches the format of issue - since it's more of a new feature rather than a bug/issue)
